birdhouse/scripts/get-components-json.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+# Obtain a JSON representation of components enabled on this platform.
+#
+# Expected result should be similar to:
+# {
+#  "components": [
+#    "bird-house/birdhouse-deploy:components/monitoring",
+#    "bird-house/birdhouse-deploy:optional-components/canarie-api-full-monitoring",
+#    "bird-house/birdhouse-deploy:optional-components/all-public-access",
+#    "bird-house/birdhouse-deploy:optional-components/wps-healthchecks",
+#    "bird-house/birdhouse-deploy:optional-components/secure-thredds",
+#    "bird-house/birdhouse-deploy:optional-components/testthredds",
+#    "bird-house/birdhouse-deploy:optional-components/test-weaver",
+#    "bird-house/birdhouse-deploy:components/weaver",
+#    "bird-house/birdhouse-deploy:components/cowbird",
+#    "custom:daccs-env"
```

Comment:
I think showing all the built-in components is fine, they are already public anyways.  I am a bit uneasy to show other custom and potentially private components.

Let me try on PAVICS staging to see what leaks out.  We do have a few private extra components.

Reply:
Sure. Let me know if you see any way to improve it easily. 

I want to avoid adding too much complexity, if possible. I don't want to impose each component to define a "display" variable, as it goes against auto-detecting which components are active. If it is absolutely necessary, a "hide me" variable could be added for individual components, but this will require doing some `cat default.env | grep ... ` on each component. 
Note that the contents of components are not sourced to avoid breaking how the stack loads them.

Reply:
It is this part that replaces `| sed -E "s|^\"((\./)?(\.\./)+)+(.+)\"|\"custom:\\4\"|g" \`.
If there is some logic in the paths to define what should be captured by the "custom" namespace, I can add it. 
Alternatively, I could limit the inclusion of components only under `birdhouse-deploy/birdhouse`.

Reply:
My path is an absolute path like `/path/to/repo/config-dir`.  What does that `sed` line above do exactly?  Any paths not starting with a dot (`.`) is replaced with `custom`?  That should have handled by absolute path normally.

Alternatively, yes, let's just limit to listing components under `birdhouse-deploy/birdhouse` since those matter most.  Listing any custom repos is more or less useless since we might not know what these custom repos/components do anyways.  This will also avoid leaking private components out.

Reply:
Ok. I'll adjust to consider only local birdhouse-deploy components. The regex was looking for something starting with at least 1 dot and any number of relative parent dirs with `../`, but it did not consider other absolute paths.

Reply:
Yes, just keep it simple and only broadcast local components for now.  Then we do not have to worry about external private components.

Reply:
The result from https://github.com/bird-house/birdhouse-deploy/pull/281/commits/1efb04bd7e5e5115a1e7da06dbe8da16183f87dc should contain only the birdhouse-deploy components.